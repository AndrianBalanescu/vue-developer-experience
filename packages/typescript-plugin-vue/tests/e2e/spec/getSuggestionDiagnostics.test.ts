import Proto from 'typescript/lib/protocol'
import { createLanguageServerForTest } from '../server'

describe('getSuggestionDiagnostics', () => {
  test('should report suggestions in a vue file', async () => {
    const server = createLanguageServerForTest()
    server.openFile('fixture-suggestion-diagnostics.vue')
    server.sendCommand(Proto.CommandTypes.SuggestionDiagnosticsSync, <
      Proto.SuggestionDiagnosticsSyncRequestArgs
    >{
      file: server.canonicalFileName('fixture-suggestion-diagnostics.vue'),
    })

    await server.close()

    expect(server.responses).toHaveLength(1)
    const response: Proto.SuggestionDiagnosticsSyncResponse =
      server.responses[0]
    expect(response.command).toBe(Proto.CommandTypes.SuggestionDiagnosticsSync)
    expect(response.body).toHaveLength(1)
    expect(
      (response.body as Proto.Diagnostic[]).map(item => [item.text, item.start])
    ).toMatchSnapshot()
  })
})
