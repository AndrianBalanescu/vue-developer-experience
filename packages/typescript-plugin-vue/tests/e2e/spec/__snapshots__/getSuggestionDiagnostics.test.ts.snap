// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSuggestionDiagnostics should report suggestions in a vue file 1`] = `
Array [
  Array [
    "'foo' is declared but its value is never read.",
    Object {
      "line": 4,
      "offset": 7,
    },
  ],
]
`;
